hashintel/hash
# Sort properties and links alphabetically in entity type editor tables

## 🌟 What is the purpose of this PR?




Sorts the properties and links tables in the entity editor alphabetically. Compensates for new rows no longer being inserted at the end by 'flashing' then in blue and, if necessary, scrolling them into view. 

**There may still be an issue with not scrolling into view when the edit bar is visible and covering the new role. I'll investigate, but will likely need to fix in a follow up**

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203543021352032/1203661700052726/f) _(internal)_

## 🔍 What does this change?




- Moved EditBar as needed to make use of `EDIT_BAR_HEIGHT` in the scroll logic for flashing table rows
- Moved the logic for resolving `link` and `property` up from the table rows to the tables themselves, as its necessary to have access to the titles of each in order to sort the rows.
	- New `sortRows` function to help sort the rows of a react hook form field array
- New `useFlashRows` hook to manage the timing of flashing and unflashing new rows in property/links tables
-  New `flash` prop on `EntityTypeTableRow`, which will flash the row and scroll it into view (if its out of view), ensuring it's not under the edit bar.

## ⚠️ Known issues



N/A

## 🐾 Next steps



N/A

## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Add a new property or link
2. See its in the right place and flashes as expected

## 📹 Demo



(Recorded before merging with main, so the type selector went off screen when it shouldn't)


https://user-images.githubusercontent.com/719814/214285367-cef6c62a-3afc-4763-9279-09afb2f89d20.mov
